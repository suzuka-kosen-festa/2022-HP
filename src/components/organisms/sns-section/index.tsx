import type { FC } from "react";
import tw from "twin.macro";
import type { SnsSectionProperties, SocialData } from "./types/model";
import { Heading } from "../../atoms/heading";
import { Text } from "../../atoms/text";
import Icon from "../../atoms/icon";

const SnsSectionContainer = tw.section`flex flex-col justify-center items-center`;

const Social = tw.div`flex space-x-8 sm:space-x-12 justify-center items-center`;

const SnsSection: FC<SnsSectionProperties> = ({ title, children, socialData, ...rest }) => (
  <SnsSectionContainer aria-label="SNS" {...rest}>
    <Heading css={tw`mb-4 sm:mb-8`} colorKey="blue">
      {title}
    </Heading>
    <Text css={tw`whitespace-pre-wrap mb-6 sm:mb-16`}>{children}</Text>
    <Social>
      {socialData.map(data => (
        <Icon
          as="a"
          css={tw`w-10 h-10 sm:(w-20 h-20) cursor-pointer`}
          icon={data.icon}
          href={data.link}
          target="_blank"
          rel="noopener noreferrer"
          iconStyles={data.style}
          key={data.key}
        />
      ))}
    </Social>
  </SnsSectionContainer>
);

export { SnsSection };
