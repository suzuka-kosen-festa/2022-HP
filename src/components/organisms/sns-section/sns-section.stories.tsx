import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { SnsSection } from ".";

type T = typeof SnsSection;
type Story = ComponentStoryObj<T>;

const data = {
  title: "SNS",
};

export default { args: { title: data.title }, component: SnsSection } as ComponentMeta<T>;

export const Default: Story = {};
