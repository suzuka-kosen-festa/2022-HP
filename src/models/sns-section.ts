import type { ComponentProps, ReactHTML } from "react";

export type SnsSectionProperties = Omit<ComponentProps<ReactHTML["section"]>, "children"> & {
  title: string;
};
