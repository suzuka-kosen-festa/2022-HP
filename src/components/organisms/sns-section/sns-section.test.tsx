import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "./sns-section.stories";

const { Default } = composeStories(stories);

const options = {
  name: "SNS",
};

describe("(components) organisms/sns-section", () => {
  test("to be organisms", () => {
    const { container } = render(<Default />);
    expect(container).toBeOrganism();
  });
  test("to be [role=region]", () => {
    const { getByRole } = render(<Default />);
    expect(getByRole("region", options));
  });
  test("take snap shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
