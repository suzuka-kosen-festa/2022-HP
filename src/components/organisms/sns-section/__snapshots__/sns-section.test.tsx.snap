// Vitest Snapshot v1

exports[`(components) organisms/sns-section > take snap shot 1`] = `
<div>
  <section
    aria-label="SNS"
    class="css-1648j2y"
  >
    <h1
      class="css-6if9cu"
    >
      SNS
    </h1>
    <div>
      Loading...
    </div>
  </section>
</div>
`;
